### ISSUE TITLE
Error trying to save table with column of Union{Int, UnitRange}

### ISSUE THREAD

**Issue by marekdedic** [JuliaData/JuliaDB.jl#331]
Hi, I'm getting this strange error trying to save a table with a column of type `Union{Int, UnitRange}`:

```julia
julia> t = table(Vector{Union{Int, UnitRange}}())
Table with 0 rows, 1 columns:
1
─

julia> save(t, "test.juliadb")
ERROR: ArgumentError: type does not have a definite number of fields
Stacktrace:
 [1] fieldcount(::Any) at ./reflection.jl:680
 [2] fixedlength(::Type, ::IdDict{Any,Any}) at /home/cisco/.julia/packages/MemPool/tlPqB/src/io.jl:156
 [3] fixedlength at /home/cisco/.julia/packages/MemPool/tlPqB/src/io.jl:146 [inlined]
 [4] mmwrite(::Serialization.Serializer{IOStream}, ::Array{Union{Int64, UnitRange},1}) at /home/cisco/.julia/packages/MemPool/tlPqB/src/io.jl:46
 [5] mmwrite(::Serialization.Serializer{IOStream}, ::StructArrays.StructArray{Tuple{Union{Int64, UnitRange}},1,Tuple{Array{Union{Int64, UnitRange},1}},Int64}) at /home/cisco/.julia/packages/JuliaDB/7cG1k/src/serialize.jl:43
 [6] mmwrite(::Serialization.Serializer{IOStream}, ::IndexedTable{StructArrays.StructArray{Tuple{Union{Int64, UnitRange}},1,Tuple{Array{Union{Int64, UnitRange},1}},Int64}}) at /home/cisco/.julia/packages/JuliaDB/7cG1k/src/serialize.jl:78
 [7] serialize(::Serialization.Serializer{IOStream}, ::MemPool.MMWrap) at /home/cisco/.julia/packages/MemPool/tlPqB/src/MemPool.jl:25
 [8] serialize at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Serialization/src/Serialization.jl:685 [inlined]
 [9] #202 at /home/cisco/.julia/packages/JuliaDB/7cG1k/src/io.jl:213 [inlined]
 [10] #open#271(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(open), ::JuliaDB.var"#202#203"{IndexedTable{StructArrays.StructArray{Tuple{Union{Int64, UnitRange}},1,Tuple{Array{Union{Int64, UnitRange},1}},Int64}}}, ::String, ::Vararg{String,N} where N) at ./io.jl:298
 [11] open at ./io.jl:296 [inlined]
 [12] save(::IndexedTable{StructArrays.StructArray{Tuple{Union{Int64, UnitRange}},1,Tuple{Array{Union{Int64, UnitRange},1}},Int64}}, ::String) at /home/cisco/.julia/packages/JuliaDB/7cG1k/src/io.jl:212
 [13] top-level scope at REPL[20]:1
 [14] eval(::Module, ::Any) at ./boot.jl:330
 [15] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/REPL/src/REPL.jl:86
 [16] run_backend(::REPL.REPLBackend) at /home/cisco/.julia/packages/Revise/AMRie/src/Revise.jl:1023
 [17] top-level scope at none:0
```

I've tried other combinations, `Union{Int, String}` does it too, `Union{Int, Float32}` doesn't...

**Comment by joshday** [comment]
I think I was the one that hacked a solution for `Union{T, Missing}` in MemPool.  I guess I didn't also cover the use case of other unions.  I'll take a look when I get a chance, but that could very well be a few weeks.

Here's where I'd start if someone else wants to fix it:
https://github.com/JuliaComputing/MemPool.jl/blob/master/src/io.jl#L38

**Comment by marekdedic** [comment]
Hi,
thanks. Just a new obervation: `VersionNumber` also causes the same issue (like, on its own, not in a Union)...

**Comment by marekdedic** [comment]
This issue is fixed as of MemPool 0.3.3 and I think it was fixed by MemPool.jl#44.
